src/effects/effectslot.cpp

```diff
@@ -178,16 +179,20 @@ EffectManifestPointer EffectSlot::getManifest() const {
     return m_pManifest;
 }
 
-void EffectSlot::reload(const QSet<ChannelHandleAndGroup>& activeInputChannels) {
-    loadEffect(m_pManifest,
-            m_pEffectsManager->createProcessor(m_pManifest),
-            activeInputChannels);
```

Comment:
perhaps instead of calling EffectSlot::loadEffect here, you could move the part of EffectSlot::loadEffect that loads the parameter slots to a new private function and call it both here and from loadEffect